### PR TITLE
=clt silence heartbeat logging in cluster client

### PR DESCRIPTION
**Pull request by ktoso**
Resolves https://github.com/akka/akka/issues/19381

### CODE REVIEW

**Comment by ktoso** [comment]
Please have a look @patriknw

File: akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala [hunk 1]
```diff
@@ -470,7 +471,12 @@ object ClusterReceptionistSettings {
 final class ClusterReceptionistSettings(
   val role: Option[String],
   val numberOfContacts: Int,
-  val responseTunnelReceiveTimeout: FiniteDuration) extends NoSerializationVerificationNeeded {
+  val responseTunnelReceiveTimeout: FiniteDuration,
+  val verboseHeartbeatLogging: Boolean) extends NoSerializationVerificationNeeded {
+
+  def this(role: Option[String], numberOfContacts: Int, responseTunnelReceiveTimeout: FiniteDuration) {
```
Comment (ktoso) [inline akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala]: for bin compat only...

File: akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala [hunk 2]
```diff
@@ -620,7 +627,7 @@ final class ClusterReceptionist(pubSubMediator: ActorRef, settings: ClusterRecep
       pubSubMediator.tell(msg, tunnel)
 
     case Heartbeat ⇒
-      log.debug("Heartbeat from client [{}]", sender().path)
+      if (verboseHeartbeatLogging) log.debug("Heartbeat from client [{}]", sender().path)
```
Comment (patriknw) [inline akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala]: In addition to this we should silence the real Cluster heartbeats also. The debug logging in this file: https://github.com/akka/akka/blob/master/akka-cluster/src/main/scala/akka/cluster/ClusterHeartbeat.scala

Therefore the configuration property should be located in akka-cluster/reference.conf, but I think we can use the same setting for the ClusterClient.

File: akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala (hunk 1 above)
Comment (patriknw) [inline akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala]: When you add the config property to ClusterSettings I think it's not necessary to mess with the ClusterReceptionistSettings, instead you can use `Cluster(context.system).settings.VerboseHeartbeatLogging directly inside the ClusterReceptionist actor.

File: akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala (hunk 2 above)
Comment (ktoso) [inline akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala]: Right, I was thinking if we want to silence both actually. Sounds good, I'll do that

**Comment by ktoso** [comment]
Updated, all cluster heartbeats debug logs are not disabled by default. Config option is `akka.cluster.debug.verbose-heartbeat-logging`

**Comment by johanandren** [comment]
LGTM!

**Comment by patriknw** [comment]
LGTM

**Pull request merged**